src/ddmd/irstate.d

```diff
@@ -53,8 +53,9 @@ struct IRState
     Dsymbols* deferToObj;           // array of Dsymbol's to run toObjFile(bool multiobj) on later
     elem* ehidden;                  // transmit hidden pointer to CallExp::toElem()
     Symbol* startaddress;
-    Array!(elem*)* varsInScope;      // variables that are in scope that will need destruction later
+    Array!(elem*)* varsInScope;     // variables that are in scope that will need destruction later
     Label*[void*]* labels;          // table of labels used/declared in function
```

Comment:
I wonder if we can change this to `Label*[void*]`, since the AA is already a reference type. This should allow to remove the `Label*[void*] labels = null;` variable declaration in https://github.com/dlang/dmd/pull/7376/files#diff-db198d44f47523b339ccc64c78f8cb53R982 and pass `null` to the constructor.

Reply:
I wanted the AA handle to be outside IRState for the same reason as varsInScope - so the destructor can be called on it there. I know there's no destructor for AAs, but I felt it should follow the same pattern as varsInScope anyway. And who knows, maybe we can replace it with a destructing AA and plug another leak.